### DISCUSSION TITLE
Can you use environment variables in fargate?

### DISCUSSION THREAD

**Discussion by LiamJBerridge** [artilleryio/artillery#2257] [Artillery]
Hi, 

I hope this isn't a duplicate quesiton, I've had a look back and I can't see one similar. 

We have an extraHTTPHeader to allow us past our WAF when loadtesting. When running the test locally you export it into the terminal and that works fine but is there a way to bundle env variables so that we can remove it from our source code and config.yml? 

Thanks in advance

**Comment by bernardobridge** [comment]
Hi @LiamJBerridge 👋 , you can set it in a `.env` file and use the `--dotenv` flag to pass the path to your tests (works locally and in Fargate).

**Reply by LiamJBerridge** [reply]
Hey @bernardobridge 👋 

So i've got the value in a .env file in the directory above the config.yml
x_origin_synthetic='SomeValue'

and in my config I have the variable 
  engines:
    playwright:
      aggregateByName: true
      extendedMetrics: true
      launchOptions:
        headless: true
      contextOptions:
        extraHTTPHeaders:
          x-origin-synthetic: '{{ $env.x_origin_synthetic }}'
          
artillery run --config config.yml scenarios/initialiseHomepage.yml  --dotenv "../.env"

The tests are still failing so our value isn't making it into the header, am I missing something?

**Reply by bernardobridge** [reply]
👋  Can you try with `$processEnvironment` instead of `$env`? I think there may be a bug in the templating logic in `config` where it isn't recognising `$env` (which replaced `$processEnvironment` recently).

**Reply by LiamJBerridge** [reply]
🙌  That worked

For some of our tests, when developing them locally we have been using inline variables like such

--variables "{\"fascia\": \"Shop1\" ,\"completePayment\": \"false\"}"

What is the process of using them with fargate?

**Reply by bernardobridge** [reply]
Hi @LiamJBerridge - to clarify, `$env` is the correct (new way) to do it. It's just that there's a bug in this particular part of the code that's not taking them into account. I've added [this Issue](https://github.com/artilleryio/artillery/issues/2258) so we can track and fix it 👍 

---

`--variables` should be available in Fargate the same way it's available locally. `run:fargate` shares that flag with the `run` command.

**Reply by LiamJBerridge** [reply]
@bernardobridge My mistake, I misread that! 

Thanks again for your help!

**Reply by LiamJBerridge** [reply]
@bernardobridge 

artillery run-fargate  --config config.yml scenarios/initialiseHomepage.yml --variables "{\"fascia\": \"Shop1\"}" --dotenv "../.env"
 Error: Nonexistent flag: --variables
 
 Same error is not produced with artillery run 😄

**Reply by bernardobridge** [reply]
Thanks for reporting that - that's a bug then @LiamJBerridge, we'll have to solve that too.

Do those variables need to be set through the CLI (i.e. dynamically)? If not, then you could include them under `variables` in your config, and that should work. So basically:

```yaml
config:
   variables:
      fascia: Shop1
   engines:
      playwright: 
       ...

```

**Reply by LiamJBerridge** [reply]
Yeah we could do, it was just nice having the ability to change it on the fly from he terminal 👍

**Reply by bernardobridge** [reply]
Just an update here @LiamJBerridge - I've [fixed this now](https://github.com/artilleryio/artillery/issues/2261). It will be fixed in the next release of Artillery.
